src/components/List/List.js

```diff
@@ -17,36 +19,37 @@ const NAVIGATION_STEP_VALUES = {
     [LIST_NAVIGATION_DIRECTIONS.DOWN]: 1,
 };
 
-const SCROLL_INTO_VIEW_SETTINGS = {
-    block: 'nearest',
-};
-
 export const NOT_LIST_CHILD = 'data-list-exception';
 export const LIST_CHILD = 'data-list-child';
 
 const List = React.forwardRef((props, ref) => {
-    const [selectedIndex, setSelectedIndex] = React.useState(null);
-    const [highlightedWithKeyboard, setHighlightedWithKeyboard] = React.useState(false);
-
-    const highlightedListItem = React.useRef(null);
-
     const { children, isDivided, doSelectOnNavigate, isControlledNavigation, ...rest } = props;
 
-    /**
-     * Scroll list if it's necessary to make the highlighted item visible
-     * every time selectedIndex was changed with the keyboard navigation
-     */
+    const [selectedIndex, setSelectedIndex] = React.useState(null);
+    const [lastNavDirection, setLastNavDirection] = React.useState(null);
+    const navigationElementRef = React.createRef();
+
+    // set selectedIndex to first selectedItem that we can find
     React.useEffect(() => {
-        const hasScrollIntoViewFunction =
-            highlightedListItem &&
-            highlightedListItem.current &&
-            highlightedListItem.current.scrollIntoView;
-
-        if (highlightedWithKeyboard && hasScrollIntoViewFunction) {
-            highlightedListItem.current.scrollIntoView(SCROLL_INTO_VIEW_SETTINGS);
-            setHighlightedWithKeyboard(false);
+        if (!children) return;
+
+        // eslint-disable-next-line no-plusplus
+        for (let i = 0; i < children.length; i++) {
+            const child = children[i];
+            if (!child.props[NOT_LIST_CHILD] && child.props.isSelected) {
+                setSelectedIndex(i);
+                break;
+            }
+        }
```

Comment:
just to avoid an additional function end point and `for` cycle we can refactor it to
```
if (children) {
    children.forEach((child, index) => {
        if (!child.props[NOT_LIST_CHILD] && child.props.isSelected) {
            setSelectedIndex(i);
        }
    });
}
```

Reply:
using forEach doesn't allow you to break the loop. So early exit is not possible.